library/src/main/java/com/mux/video/upload/internal/UploadMetrics.kt

```diff
@@ -13,14 +13,191 @@ import okhttp3.Interceptor
 import okhttp3.MediaType.Companion.toMediaType
 import okhttp3.Request
 import okhttp3.RequestBody.Companion.toRequestBody
+import org.json.JSONArray
 import org.json.JSONObject
+import java.text.SimpleDateFormat
 import java.util.*
 import java.util.concurrent.TimeUnit
 
+
 internal class UploadMetrics private constructor() {
 
   private val logger get() = MuxUploadSdk.logger
 
+
+  private fun formatMilliseconds(ms:Long):String {
+    return String.format("%02d:%02d:%02d",
+      TimeUnit.MILLISECONDS.toHours(ms),
+      TimeUnit.MILLISECONDS.toMinutes(ms) -
+              TimeUnit.HOURS.toMinutes(TimeUnit.MILLISECONDS.toHours(ms)), // The change is in this line
+      TimeUnit.MILLISECONDS.toSeconds(ms) -
+              TimeUnit.MINUTES.toSeconds(TimeUnit.MILLISECONDS.toMinutes(ms)));
+  }
+
+  private suspend fun getEventInfo(startTimeMillis: Long,
+                                   endTimeMillis: Long,
+                                   inputFileDurationMs: Long,
+                                   uploadInfo: UploadInfo): JSONObject {
+    val iso8601Sdf = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ", Locale.getDefault())
+    var videoDuration = inputFileDurationMs
+    if (inputFileDurationMs <= 0) {
+      videoDuration = withContext(Dispatchers.IO) {
+        try {
+          MediaMetadataRetriever().use { retriever ->
+            retriever.setDataSource(uploadInfo.inputFile.absolutePath)
+            retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION)?.toInt()
+          }
+        } catch (e: Exception) {
+          MuxUploadSdk.logger.e("UploadMetrics", "Failed to get video duration", e)
+          null
+        }
+      }!!.toLong()
+    }
+    return JSONObject().apply {
```

Comment:
issue(blocker): These are still not consistent with the documentation. A few high level callouts:

* Different event/POST `"types"` will have different `data` body keys/values (See docs example JSON for e.g. `upload_succeeded` vs. `upload_input_standardization_succeeded`
* Some of these keys are still incorrect (See, e.g. `platform_name` vs. `os_name`)
* Some of these values are are still not consistent with docs (e.g. `input_duration`, which should be numeric seconds).

@andrewjl-mux provided some feedback in a thread on slack, but let me know if you need additional direction or documentation.